core/Enclave/enclave_hsm.edl

```diff
@@ -67,82 +67,78 @@ enclave {
-                                            [out, size=ciphertext_len] uint8_t *ciphertext, uint32_t ciphertext_len);
-        public sgx_status_t enclave_rsa_decrypt([in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                                            [in, size=ciphertext_len] const uint8_t *ciphertext, uint32_t ciphertext_len,
-                                            [out, size=plaintext_len] uint8_t *plaintext, uint32_t plaintext_len,
-                                            [out] uint32_t *req_plaintext_len);
+        public sgx_status_t enclave_create_key([in, out, size=cmk_len] ehsm_keyblob_t *cmk, size_t cmk_len);
+
+        public sgx_status_t enclave_encrypt([in, size=cmk_len] const ehsm_keyblob_t* cmk, size_t cmk_len,
+                            [in, size=aad_len] const ehsm_data_t *aad, size_t aad_len,
+                            [in, size=plaintext_len] const ehsm_data_t *plaintext, size_t plaintext_len,
+                            [in, out, size=ciphertext_len] ehsm_data_t *ciphertext, size_t ciphertext_len);
+
+        public sgx_status_t enclave_decrypt([in, size=cmk_len] const ehsm_keyblob_t* cmk, size_t cmk_len,
+                            [in, size=aad_len] const ehsm_data_t *aad, size_t aad_len,
+                            [in, size=ciphertext_len] const ehsm_data_t *ciphertext, size_t ciphertext_len,
+                            [in, out, size=plaintext_len] ehsm_data_t *plaintext, size_t plaintext_len);
+
+        public sgx_status_t enclave_asymmetric_encrypt([in, size=cmk_len] const ehsm_keyblob_t* cmk, size_t cmk_len,
+                            [in, size=plaintext_len] ehsm_data_t *plaintext, size_t plaintext_len,
+                            [in, out, size=ciphertext_len] ehsm_data_t *ciphertext, size_t ciphertext_len);
+
+        public sgx_status_t enclave_asymmetric_decrypt([in, size=cmk_len] const ehsm_keyblob_t* cmk, size_t cmk_len,
+                            [in, size=ciphertext_len] ehsm_data_t *ciphertext, uint32_t ciphertext_len,
+                            [in, out, size=plaintext_len] ehsm_data_t *plaintext, uint32_t plaintext_len);
+
+        public sgx_status_t enclave_sign([in, size=cmk_len] const ehsm_keyblob_t* cmk, size_t cmk_len,
+                            [in, size=data_len] const ehsm_data_t *data, size_t data_len,
+                            [in, size=appid_len] const ehsm_data_t *appid, size_t appid_len,
+                            [in, out, size=signature_len] ehsm_data_t *signature, size_t signature_len);
+                                         
+        public sgx_status_t enclave_verify([in, size=cmk_len] const ehsm_keyblob_t* cmk, size_t cmk_len,
+                            [in, size=data_len] const ehsm_data_t *data, size_t data_len,
+                            [in, size=appid_len] const ehsm_data_t *appid, size_t appid_len,
```

Comment:
https://github.com/guanzhi/GM-Standards/blob/master/GMT%E6%AD%A3%E5%BC%8F%E6%A0%87%E5%87%86/GMT%200009-2012%20SM2%E5%AF%86%E7%A0%81%E7%AE%97%E6%B3%95%E4%BD%BF%E7%94%A8%E8%A7%84%E8%8C%83%20.pdf
please refer to chapter 10.
user id for SM2 can be set to 1234567812345678 by default, like this:
unsigned char sm2_user_id[] = "1234567812345678";
unsigned int sm2_user_id_len = sizeof(sm2_user_id)-1;

Please remove all appid related codes in enclave.

Reply:
Does it mean that don't need to input the appid as an ID for signature, and use a default value for signature

Reply:
yes.   define the default value inside enclave.